zone/client_packet.cpp

```diff
@@ -16542,3 +16697,300 @@ void Client::ReloadExpansionProfileSetting()
 		m_pp.expansions = RuleI(World, ExpansionSettings);
 	}
 }
+
+void Client::Handle_OP_GuildTributeSelect(const EQApplicationPacket *app)
+{
+	if (app->size != sizeof(GuildTributeSelectReq_Struct)) {
+		LogError("Invalid size on OP_GuildSelectTribute packet");
+		return;
+	}
+
+	GuildTributeSelectReq_Struct *t = (GuildTributeSelectReq_Struct *) app->pBuffer;
+	SendGuildTributeDetails(t->tribute_id, t->tier);
+	return;
+}
+
+void Client::Handle_OP_GuildTributeModifyBenefits(const EQApplicationPacket *app)
+{
+	if (app->size != sizeof(GuildTributeModifyBenefits_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildModifyBenefits packet");
+		return;
+	}
+
+	GuildTributeModifyBenefits_Struct *t = (GuildTributeModifyBenefits_Struct *) app->pBuffer;
+
+	EQApplicationPacket               *outapp = new EQApplicationPacket(
+		OP_GuildModifyBenefits,
+		sizeof(GuildTributeModifyBenefits_Struct));
+	GuildTributeModifyBenefits_Struct *gmbs   = (GuildTributeModifyBenefits_Struct *) outapp->pBuffer;
+
+	switch (t->command) {
+		case GUILD_TRIBUTES_SAVE: {
+			gmbs->command           = GUILD_TRIBUTES_SAVE;
+			gmbs->data              = 1;
+			gmbs->tribute_id_1      = t->tribute_id_1;
+			gmbs->tribute_id_2      = t->tribute_id_2;
+			gmbs->tribute_id_1_tier = t->tribute_id_1_tier;
+			gmbs->tribute_id_2_tier = t->tribute_id_2_tier;
+			gmbs->tribute_master_id = t->tribute_master_id;
+			QueuePacket(outapp);
+
+			auto sp  = new ServerPacket(ServerOP_RequestGuildActiveTributes, sizeof(GuildTributeUpdate));
+			auto out = (GuildTributeUpdate *) sp->pBuffer;
+			out->guild_id = GuildID();
+			out->enabled  = false;
+			worldserver.SendPacket(sp);
+			safe_delete(sp);
+			break;
+		}
+		case GUILD_TRIBUTES_MODIFY: {
+			gmbs->command           = GUILD_TRIBUTES_MODIFY;
+			gmbs->data              = 1;
+			gmbs->tribute_id_1      = t->tribute_id_1;
+			gmbs->tribute_id_2      = t->tribute_id_2;
+			gmbs->tribute_id_1_tier = t->tribute_id_1_tier;
+			gmbs->tribute_id_2_tier = t->tribute_id_2_tier;
+			gmbs->tribute_master_id = t->tribute_master_id;
+			QueuePacket(outapp);
+			break;
+		}
+		default: {
+			gmbs->command           = GUILD_TRIBUTES_SAVE;
+			gmbs->data              = 1;
+			gmbs->tribute_id_1      = 0;
+			gmbs->tribute_id_2      = 0;
+			gmbs->tribute_id_1_tier = 0;
+			gmbs->tribute_id_2_tier = 0;
+			gmbs->tribute_master_id = 0;
+			QueuePacket(outapp);
+			break;
+		}
+	}
+	safe_delete(outapp)
+	return;
```

Comment:
return at the end of `void` function isn't necessary, think I've seen this in a couple other functions